### PR TITLE
Added a recipe for pytransit.

### PR DESCRIPTION
**Pull request by hpparvi**
This PR adds a recipe for the pytransit package for exoplanet transit modelling.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/pytransit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{% set name = "pytransit" %}
+{% set version = "2.6.11" %}
+{% set python_min = "3.10" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pytransit-{{ version }}.tar.gz
```
Comment (mwcraig) [inline recipes/pytransit/meta.yaml:10]:
```suggestion
  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/pytransit-{{ version }}.tar.gz
```

File: recipes/pytransit/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  sha256: 8a5ec1a4515ff86157156f1e9571c8d4ff7bad671a46181cd8b12ecc9831020f
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python >={{ python_min }}
```
Comment (mwcraig) [inline recipes/pytransit/meta.yaml:20]:
```suggestion
    - python {{ python_min }}
```

File: recipes/pytransit/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    - pytransit
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  summary: Fast and painless exoplanet transit light curve modelling.
+  license: GPL-2.0-only
+  license_file: LICENSE
```
Comment (mwcraig) [inline recipes/pytransit/meta.yaml:59]:
```suggestion
  license_file: LICENSE
  home: https://github.com/hpparvi/PyTransit
```

File: recipes/pytransit/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    - celerite
+    - pyrr
+    - meepmeep
+
+test:
+  imports:
+    - pytransit
+  commands:
+    - pip check
+  requires:
```
Comment (mwcraig) [inline recipes/pytransit/meta.yaml:53]:
```suggestion
  requires:
    - python {{ python_min }}
```

**Review by mwcraig — CHANGES_REQUESTED** [review]
Thanks for opening this, @hpparvi! Just a few minor suggestions and it should be ready to go.

**Comment by hpparvi** [comment]
Many thanks for your comments @mwcraig! The recipe should now be ready for review.

**Comment by hpparvi** [comment]
@conda-forge/help-python, ready for review!

**Review by mwcraig — APPROVED** [review]

**Pull request merged**